spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java

```diff
@@ -435,7 +435,8 @@ public void initialize() {
 		Collection<Collection> collections = this.applicationContext.getBeansOfType(Collection.class, false, false)
 				.values();
 		for (Collection<?> collection : collections) {
-			if (collection.size() > 0 && collection.iterator().next() instanceof Declarable) {
+			//NullPointerException
+			if (collection != null && collection.size() > 0 && collection.iterator().next() instanceof Declarable) {
```

Comment:
@ooooow Could you please elaborate update what circumstances `getBeansOfType()` can return `null` bean instances?

Reply:
Suppose there is a bean below:
@Bean
    public List<String> initList(){
        List<String> stringList = null;
        if(1==2){
            stringList.add("1111");
        }
        return stringList;
    }

Reply:
if(1==2)  is a  example

Reply:
It is invalid to define a `null` bean